lib/util/git/index.ts

```diff
@@ -520,7 +520,9 @@ export async function checkoutBranch(
   logger.debug(`Setting current branch to ${branchName}`);
   await syncGit();
   try {
-    await gitRetry(() => git.checkout(['-f', branchName, '--']));
+    await gitRetry(() =>
+      git.checkout(['-f', '--recurse-submodules', branchName, '--']),
```

Comment:
what happens if submodules are not initialized?

Reply:
If there is no submodules in the repo, this flag does nothing. Just try it in a submodule-free repo.

We could make this conditional on cloneSubmodules but I doubt it's necessary.

Reply:
yes, make it conditional on `cloneSubmodules`

Reply:
otherwise it will cause submodules clone

Reply:
mmmh indeed.